checker/src/org/checkerframework/checker/index/upperbound/UBQualifier.java

```diff
@@ -287,14 +289,39 @@ public boolean isLessThanLengthOf(String sequence) {
          * AnnotationMirrors using @{@link LTEqLengthOf} or @{@link LTOMLengthOf} are returned.
          * Otherwise, @{@link LTLengthOf} is used.
          *
-         * <p>The annotation is sorted by sequence and then offset. This is so that {@link
-         * AnnotationUtils#areSame(AnnotationMirror, AnnotationMirror)} returns true for equivalent
-         * annotations.
+         * <p>The returned annotation is canonicalized by sorting its arguments by sequence and then
+         * offset. This is so that {@link AnnotationUtils#areSame(AnnotationMirror,
+         * AnnotationMirror)} returns true for equivalent annotations.
          *
          * @param env ProcessingEnvironment
          * @return the AnnotationMirror that represents this qualifier
          */
-        public AnnotationMirror convertToAnnotationMirror(ProcessingEnvironment env) {
+        public AnnotationMirror convertToAnnotation(ProcessingEnvironment env) {
+            return convertToAnnotation(env, false);
+        }
+
+        /**
+         * Returns the @{@link SubstringIndexFor} AnnotationMirror that represents this qualifier in
```

Comment:
The returned AnnotationMirror doesn't represent this qualifier.  Maybe it's an overapproximation to it, or maybe there is some other relationship.  Please clarify what this does.
(It can't be an overapproximation, because @SubstringIndexFor forbids some values that @LTLengthOf permits.  Maybe there is some other invariant about how the values are used that makes this sound, such as the fact that this method is only ever called from SubstringIndexAnnotatedTypeFactory.convertUBQualifierToAnnotation, and that should only be called on annotations that have been created from @SubstringIndexFor annotations?  In any event, please explain those tricky bits of code.)